src/main/java/org/mockito/internal/stubbing/answers/ThrowsException.java

```diff
@@ -14,27 +14,35 @@
 import static org.mockito.internal.exceptions.Reporter.cannotStubWithNullThrowable;
 import static org.mockito.internal.exceptions.Reporter.checkedExceptionInvalid;
 
+/**
+ * An answer that always throws the same throwable.
+ */
 public class ThrowsException implements Answer<Object>, ValidableAnswer, Serializable {
 
     private static final long serialVersionUID = 1128820328555183980L;
     private final Throwable throwable;
     private final ConditionalStackTraceFilter filter = new ConditionalStackTraceFilter();
 
+    /**
+     * Creates a new answer always throwing the given throwable. If it is null,
+     * {@linkplain ValidableAnswer#validateFor(InvocationOnMock) answer validation}
+     * will fail.
+     */
     public ThrowsException(Throwable throwable) {
         this.throwable = throwable;
     }
 
     public Object answer(InvocationOnMock invocation) throws Throwable {
-        if (MockUtil.isMock(throwable)) {
-            throw throwable;
+        if (throwable == null) {
+            throw new IllegalStateException("throwable is null: " +
+                "you shall not call #answer if #validateFor fails!");
         }
-        Throwable t = throwable.fillInStackTrace();
-
-        if (t == null) {
+        if (MockUtil.isMock(throwable)) {
             throw throwable;
         }
-        filter.filter(t);
-        throw t;
+        throwable.fillInStackTrace();
```

Comment:
I'd rather keep the assignment for backwards compatibility. I made this change and will push it back to your pull request. Can you contribute a test that preserves this behavior?

Reply:
Oh, that's unexpected, thank you for adding a clarifying comment!

I think I'll add a couple of tests:
- The one that prevents the change I did.
- The one that verifies we get the throwable returned from `fillInStackTrace` in subsequent invocations.

Reply:
What's interesting is that the custom exception from #866 breaks the [Throwable#fillInStackTrace contract](https://docs.oracle.com/javase/9/docs/api/java/lang/Throwable.html#fillInStackTrace--): this method shall always return `this`, never `null` or a reference to another throwable.